pkg/bmc/redfish.go

```diff
@@ -0,0 +1,92 @@
+package bmc
+
+import (
+	"net"
+	"strings"
+)
+
+func init() {
+	registerFactory("redfish", newRedfishAccessDetails)
+	registerFactory("redfish+http", newRedfishAccessDetails)
+	registerFactory("redfish+https", newRedfishAccessDetails)
+}
+
+func newRedfishAccessDetails(bmcType, portNum, hostname, path string) (AccessDetails, error) {
+	// If the hostname is an ipv6 address, it needs to be in square brackets
+	// as we are forming a URL out of it.
+	addresses, err := net.LookupIP(hostname)
+	if err == nil {
+		for _, address := range addresses {
+			if address.String() == hostname && address.To4() == nil {
+				hostname = strings.Join([]string{"[",hostname,"]"}, "")
```

Comment:
What happens here if a host has multiple IPv6 addresses?

Reply:
Why do we even resolve the address?

Reply:
Resolving the hostname is to find out if the domain name given is an ipv6 (that will need brackets around) or not. If we are given a domain name and the host has multiple addresses, then none will match the domain name. If we are given an IPv6, the resolved IP will match the given IP and the brackets will be added.

Reply:
OK, the purpose of that loop was not obvious to me at all, even with the comment.

I think if we take the other PR that changes the access API to use a parsed URL, we would not need to perform this check here, right?

Reply:
Correct, with that other PR you could just use `.Host` instead of `.Hostname()` and you get the unmodified host+port.
This seems like a good reason to accept that PR.

Reply:
Ok, so we'll first focus on URL PR and then modify this one